scene/gui/rich_text_label.cpp

```diff
@@ -3929,6 +3930,11 @@ void RichTextLabel::_bind_methods() {
 
 void RichTextLabel::set_visible_characters(int p_visible) {
 	visible_characters = p_visible;
+	int total_char_count = get_total_character_count();
```

Comment:
Probably worth looking into caching this instead of calculating every time, like the Label class does.